pkg/operator/ceph/cluster/cluster.go

```diff
@@ -52,6 +58,68 @@ func newCluster(c *cephv1beta1.Cluster, context *clusterd.Context) *cluster {
 		ownerRef: ClusterOwnerRef(c.Namespace, string(c.UID))}
 }
 
+func (c *cluster) detectCephMajorVersion() error {
+	// get the major ceph version by running "ceph --version" in the ceph image
+	job := &batch.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      detectVersionName,
+			Namespace: c.Namespace,
+		},
+		Spec: batch.JobSpec{
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"job": detectVersionName,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Command: []string{"ceph"},
+							Args:    []string{"--version"},
+							Name:    "version",
+							Image:   c.Spec.CephVersion.Image,
+						},
+					},
+					RestartPolicy: v1.RestartPolicyOnFailure,
+				},
+			},
+		},
+	}
+	k8sutil.SetOwnerRef(c.context.Clientset, c.Namespace, &job.ObjectMeta, &c.ownerRef)
+
+	// run the job to detect the version
+	if err := k8sutil.RunReplaceableJob(c.context.Clientset, job); err != nil {
+		return fmt.Errorf("failed to start version job. %+v", err)
+	}
+
+	if err := k8sutil.WaitForJobCompletion(c.context.Clientset, job); err != nil {
+		return fmt.Errorf("failed to complete version job. %+v", err)
```

Comment:
If the job fails to complete, what does this mean? Is the job container running indefinitely?

Reply:
Either job is still running after the timeout (15 minutes), or the job wasn't found.